src/tracers.jl

```diff
@@ -200,13 +206,16 @@ end
 #===========#
 
 myempty(::Type{ConnectivityTracer{I}}) where {I} = ConnectivityTracer{I}(myempty(I), true)
-myempty(::Type{GradientTracer{G}}) where {G}     = GradientTracer{G}(myempty(G), true)
-myempty(::Type{HessianTracer{G,H}}) where {G,H}  = HessianTracer{G,H}(myempty(G), myempty(H), true)
+myempty(::Type{GradientTracer{G}}) where {G} = GradientTracer{G}(myempty(G), true)
```

Comment:
Why not support shared `Gradient`- and `ConnectivityTracer`s?

Reply:
What's interesting about the shared Hessian is that the gradient part of each `HessianTracer` can still vary

Reply:
So in #119 lingo, you have a `SetVectorPattern` and a `SharedMatrixPattern`. 😉

I hope you see the point of the added pattern abstraction: it's a nuisance to make these changes on the three tracer types in almost every PR.

Reply:
Why would we ever need a shared anything for `ConnectivityTracer` or `GradientTracer`?
If you share the underlying storage, every output variable has the same gradient pattern, so your Jacobian has constant rows (not columns, sorry).